server.js

```diff
@@ -268,30 +267,37 @@ app.use("/site/:siteId", async (req, res, next) => {
     const headObjectResponse = await s3Handler.send(headCommand);
 
     // Set the Content-Type based on the file extension
-    const fileExtension = path.extname(filePath).toLowerCase();
-    if (fileExtension === ".html") {
-      res.set("Content-Type", "text/html");
-    } else {
-      res.set("Content-Type", headObjectResponse.ContentType);
-    }
+    const fileExtension = path.extname(filePath).substring(1);
+    const mimeTypes = {
+      html: "text/html",
+      css: "text/css",
+      js: "application/javascript",
+      json: "application/json",
+      png: "image/png",
+      jpg: "image/jpeg",
+      jpeg: "image/jpeg",
+      svg: "image/svg+xml",
+      gif: "image/gif",
+    };
+    const contentType = mimeTypes[fileExtension] || "application/octet-stream";
+    res.setHeader("Content-Type", contentType);
 
-    const getCommand = new GetObjectCommand(params);
-    const { Body } = await s3Handler.send(getCommand);
-    Body.pipe(res);
+    const data = await s3Handler.send(new GetObjectCommand(params));
+    data.Body.pipe(res);
   } catch (error) {
-    console.error(`Error fetching ${key}: ${error}`);
     if (error.name === "NoSuchKey") {
-      console.log(`File not found: ${key}`);
-      return next();
+      res.status(404).send("File not found");
+    } else {
+      next(error);
     }
-    res.status(500).send("An error occurred");
   }
 });
 
-handleOnboardingSocketEvents(io);
-
-const PORT = process.env.PORT || 5001;
+io.on("connection", (socket) => {
+  handleOnboardingSocketEvents(socket);
+});
 
+const PORT = process.env.PORT || 4000;
 server.listen(PORT, () => {
-  console.log(`Server running at ${PORT}`);
+  console.log(`Server is running on port ${PORT}`);
```

Comment:
interesting rephrase, can you explain the logic behind this?
Does the server run on port or at port?

Reply:
just thought to go with some of my terms, sorry for that, "at" will do too..